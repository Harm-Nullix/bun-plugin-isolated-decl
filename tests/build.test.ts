import path from 'node:path';
import { $ } from 'bun';
import { expect, test } from 'bun:test';
import dts from '../index.ts';

test('build', async () => {
	const input = path.resolve(__dirname, 'fixtures/main.ts');
	const dist = path.resolve(__dirname, 'out');

	await $`rm -rf ${dist}`;

	await Bun.build({
		entrypoints: [input],
		outdir: dist,
		minify: false,
		plugins: [dts()],
		target: 'bun',
		external: ['*'],
	});

	const outputDts = await $`cat ${path.resolve(dist, 'main.d.ts')}`.text();
	expect(outputDts).toMatchSnapshot();
});
