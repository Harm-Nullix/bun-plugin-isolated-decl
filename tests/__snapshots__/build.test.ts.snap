// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`build 1`] = `
"export type Str = string;
export declare function hello(s: Str): Str;
"
`;

exports[`build with glob path 1`] = `
"export type Str = string;
export declare function hello(s: Str): Str;
"
`;
